course3/homework-unique_ptr/unique_ptr.cc

```diff
@@ -0,0 +1,111 @@
+#include "unique_ptr.h"
+
+#include <utility>
+
+
+template<class T>
+karun::unique_ptr<T>::unique_ptr() : mPtr(nullptr) {}
```

Comment:
There shouldn't be any. And it's fine to set the value to nullptr in class definition. The value in class definition is the one used during default construction of the member; in you other constructors you perform non-default initialization.